inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -250,31 +248,72 @@ def _author_list(obj, eng):
-                    try:
-                        authors_xml = convert(xml_content, stylesheet)
-                    except XMLSyntaxError:
-                        # Probably the %auto-ignore comment exists, so we skip the
-                        # first line. See: inspirehep/inspire-next/issues/2195
-                        authors_xml = convert(
-                            xml_content.split('\n', 1)[1],
-                            stylesheet,
+
+                if isinstance(xml_content, str):
+                    xml_content = xml_content.decode('utf-8')
+
+                # Probably the %auto-ignore comment exists, so we skip the
+                # first line. See: inspirehep/inspire-next/issues/2195
+                if "%auto-ignore" in xml_content:
+                    content = Selector(text=xml_content.split('\n', 1)[1], type="xml")
+                else:
+                    content = Selector(text=xml_content, type="xml")
+
+                content.remove_namespaces()
+                # Going through all the authors in the file
+                for author in content.xpath("//Person"):
+
+                    ids = []
+                    affiliations = []
+                    # Checks which path is the correct one for this file
+                    # to get author ids
+                    if author.xpath("./authorIDs/authorID") == []:
+                        index = 1
+                    else:
+                        index = 0
+
+                    # Getting all the author ids
+                    for id in author.xpath(author_id_paths[index]):
+                        if id.xpath("./@source").get() == u'CCID':
+                            ids.append(['CERN', id.xpath("./text()").get()])
+                        elif id.xpath("./@source").get() == u'INSPIRE':
+                            ids.append([id.xpath("./@source").get() + ' ID', id.xpath("./text()").get()])
+                        else:
+                            ids.append([id.xpath("./@source").get(), id.xpath("./text()").get()])
+
+                    # Getting all the names for affiliated organizations using the organization ids from author
+                    for affiliation in author.xpath("./authorAffiliations/authorAffiliation/@organizationid").getall():
+                        affiliations.append(content.xpath("//organizations/Organization[@id='" + affiliation + "']/orgName[@source='spiresICN' or @source='INSPIRE']/text()").get())
+
+                    name = normalize_name(author.xpath(".//familyName/text()").get() + ", " + author.xpath(".//givenName/text()").get())
+
+                    # building the info to a correct format with litratureBuilder()
+                    builder.add_author(
+                        builder.make_author(
+                            full_name=name,
+                            affiliations=affiliations,
+                            ids=ids,
                         )
+                    )
```

Comment:
could you maybe extract this part into a function that takes a string containing the XML and returns the JSON translation? that way it's easier to test and we can more easily run it manually and reuse it.